mono/metadata/sre.c

```diff
@@ -4272,6 +4273,7 @@ ensure_complete_type (MonoClass *klass, MonoError *error)
 		// Asserting here could break a lot of code
 		//g_assert (klass->wastypebuilder);
 	}
+#endif
```

Comment:
i'm a bit worried we'll never return to fix up this function once we're working on SRE.  maybe move the `#ifdef` under the `if` and add an `#else` case:

```
if (image_is_dynamic (...) && ...) {
#ifndef ENABLE_NETCORE
  ...
#else
  g_assert_not_reached ();
#endif
}
```

Reply:
Hmm, fair. Will do.